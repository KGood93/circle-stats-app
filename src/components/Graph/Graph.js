import React, { PureComponent } from 'react';
import {LineChart, Line, XAxis, YAxis, Text, Tooltip, ResponsiveContainer} from 'recharts';
import './Graph.css'

export default class Graph extends PureComponent {
  //static jsfiddleUrl = 'https://jsfiddle.net/alidingling/xqjtetw0/';
  
  render() {
    const data = this.props.workingData.data
    //console.log(data)

    return (
      <ResponsiveContainer width="100%" height={300}>
        <LineChart
          data={data}
          margin={{top: 5, right: 30, left: 20, bottom: 5}}
        >

          <XAxis dataKey="date" />
          <YAxis dataKey="attendance" 
            label={<Text x={0} y={0} dx={50} dy={150} offset={0} angle={-90}>Attendance</Text>}
            />
          <Tooltip />
          <Line type="monotone" dataKey="attendance" stroke="#0c465e" activeDot={{ r: 8 }} />
        </LineChart>
      </ResponsiveContainer>
    );
  }
}