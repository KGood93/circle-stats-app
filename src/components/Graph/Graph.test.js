import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Graph from './Graph';

describe('Graph Component', () => {
    it(`renders without crashing`, () => {
        const value = {data: [
            {
                date: '2019-12-25',
                location: 'Parents House',
                attendance: 24,
                notes: 'Christmas',
            },
            {
                date: '2019-11-28',
                location: 'London',
                attendance: 12,
                notes: 'Thanksgiving',
            },
        ]}
        const wrapper = shallow(<Graph workingData={value}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});