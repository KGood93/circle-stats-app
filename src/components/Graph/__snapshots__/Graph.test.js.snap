// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph Component renders without crashing 1`] = `
<ResponsiveContainer
  debounce={0}
  height={300}
  width="100%"
>
  <LineChart
    barCategoryGap="10%"
    barGap={4}
    data={
      Array [
        Object {
          "attendance": 24,
          "date": "2019-12-25",
          "location": "Parents House",
          "notes": "Christmas",
        },
        Object {
          "attendance": 12,
          "date": "2019-11-28",
          "location": "London",
          "notes": "Thanksgiving",
        },
      ]
    }
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 20,
        "right": 30,
        "top": 5,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
  >
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="date"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      stroke="black"
      tickCount={5}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      allowDuplicatedCategory={true}
      dataKey="attendance"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      label={
        <Text
          angle={-90}
          capHeight="0.71em"
          dx={50}
          dy={175}
          lineHeight="1em"
          offset={0}
          scaleToFit={false}
          textAnchor="start"
          verticalAnchor="end"
          x={0}
          y={0}
        >
          Attendance
        </Text>
      }
      mirror={false}
      orientation="left"
      padding={
        Object {
          "bottom": 0,
          "top": 0,
        }
      }
      reversed={false}
      scale="auto"
      stroke="black"
      tickCount={5}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Tooltip
      active={false}
      allowEscapeViewBox={
        Object {
          "x": false,
          "y": false,
        }
      }
      animationDuration={400}
      animationEasing="ease"
      contentStyle={Object {}}
      coordinate={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      cursor={true}
      cursorStyle={Object {}}
      filterNull={true}
      isAnimationActive={true}
      itemStyle={Object {}}
      labelStyle={Object {}}
      offset={10}
      separator=" : "
      useTranslate3d={false}
      viewBox={
        Object {
          "x1": 0,
          "x2": 0,
          "y1": 0,
          "y2": 0,
        }
      }
      wrapperStyle={Object {}}
    />
    <Line
      activeDot={
        Object {
          "r": 8,
        }
      }
      animateNewValues={true}
      animationBegin={0}
      animationDuration={1500}
      animationEasing="ease"
      connectNulls={false}
      dataKey="attendance"
      dot={true}
      fill="#fff"
      hide={false}
      isAnimationActive={true}
      legendType="line"
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      points={Array []}
      stroke="#0c465e"
      strokeWidth="2"
      type="monotone"
      xAxisId={0}
      yAxisId={0}
    />
  </LineChart>
</ResponsiveContainer>
`;
